### PR TITLE
feat: create a confirmation modal on mint items from collection

### LINKED ISSUE #1566: Create a confirmation step for the "MintItemsModal"
Create a confirmation step after the user clicks on "mint" in the MintItemsModals that shows a description of the items that includes the beneficiaries and the amount of each item to be minted. This step should also include the total (in numeric form) of the items that are going to be minted.

### PR DESCRIPTION
**Pull request by flobarreto**
closes #1566

### CODE REVIEW

**Comment by flobarreto** [comment]
image

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 1]
```diff
@@ -139,14 +194,42 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
+                {isEmpty ? (
+                  <Button secondary fluid onClick={onClose}>
+                    {t('global.cancel')}
+                  </Button>
+                ) : (
+                  <NetworkButton
+                    primary
+                    onClick={() => this.setState({ confirm: true })}
+                    loading={isLoading}
+                    disabled={isDisabled || !!error}
```
Comment (LautaroPetaccio) [inline src/components/Modals/MintItemsModal/MintItemsModal.tsx:219]: If I'm not wrong the `!!error` condition shouldn't be here, but in the confirm button, right?

File: src/modules/translation/languages/en.json [hunk 2]
```diff
@@ -1366,7 +1366,11 @@
     "add_item": "Add Item",
     "limit_reached": "You can only mint {max} items per transaction",
     "no_items": "You have no items available to mint for \"{name}\"",
-    "unsynced_warning": "Some items have changes that have not been approved yet."
+    "unsynced_warning": "Some items have changes that have not been approved yet.",
+    "next": "Next",
+    "total_items": "Total items to be minted",
+    "confirm_title":  "You are about to mint {amount} items, each item will be minted to the given address with the provided amount.",
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/en.json:1372]: There's a way to write different cases of plurality differently, for example, if there's one item to be minted, we can write "item" and when there are several, we can write "items":
```suggestion
    "confirm_title":  "You are about to mint {amount} {count, plural, one {item} other {items}}, each item will be minted to the given address with the provided amount.",
```

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

**Comment by LautaroPetaccio** [comment]
There seems to be a difference in styling when there are multiple addresses used to mint an item:
Screen Shot 2022-11-21 at 12 12 30
Screen Shot 2022-11-21 at 12 12 23
Would you mind checking it as well?

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 3]
```diff
@@ -139,14 +194,42 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
+                  >
+                    {t('mint_items_modal.next')}
+                  </NetworkButton>
+                )}
+              </ModalActions>
+              {error ? (
+                <Row className="error" align="right">
+                  <p className="danger-text">{error}</p>
+                </Row>
+              ) : null}
```
[the remark this answers is not in the document]
Comment (flobarreto) [reply src/components/Modals/MintItemsModal/MintItemsModal.tsx:230]: This was the original code not changed for this pr

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 4]
```diff
@@ -86,50 +91,100 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
+
+              <Row className="actions" align="right">
+                <Button
+                  className="back"
+                  secondary
+                  onClick={() =>
+                    this.setState(prevState => {
+                      return { ...prevState, confirm: false }
+                    })
+                  }
```
[the remark this answers is not in the document]
Comment (flobarreto) [reply src/components/Modals/MintItemsModal/MintItemsModal.tsx:183]: created a handleView to resuse it on next button

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 5]
```diff
@@ -86,50 +95,98 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
 
+    const amountItemsValue = Object.values(itemMints).reduce((accumulator, mints) => {
+      const totalMints = mints.reduce((acc, mint) => {
+        const amount = mint.amount || 0
+        return acc + amount
+      }, 0)
+      return accumulator + totalMints
+    }, 0)
+
+    // ! this function goes inside render to retriger rendering of the item dropdown when the selection of items changes
```
Comment (LautaroPetaccio) [inline src/components/Modals/MintItemsModal/MintItemsModal.tsx:121]:
```suggestion
    // ! this function goes inside render to re-trigger the rendering of the item dropdown when the selection of items changes
```

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 6]
```diff
@@ -8,11 +10,16 @@ import { canMintItem, MAX_NFTS_PER_MINT } from 'modules/item/utils'
 import { Item } from 'modules/item/types'
 import { Mint } from 'modules/collection/types'
 import ItemDropdown from 'components/ItemDropdown'
+import ItemImage from 'components/ItemImage'
 import { Props, State, ItemMints } from './MintItemsModal.types'
 import MintableItem from './MintableItem'
-
 import './MintItemsModal.css'
 
+export enum View {
```
Comment (LautaroPetaccio) [inline src/components/Modals/MintItemsModal/MintItemsModal.tsx:18]: Would you mind moving the `View` enum type to types?

File: src/modules/translation/languages/es.json [hunk 7]
```diff
@@ -1392,7 +1392,11 @@
     "add_item": "Agregar Item",
     "limit_reached": "Solo puede crear {max} items por transaction",
     "no_items": "No tienes items disponibles para crear en la colleción \"{name}\"",
-    "unsynced_warning": "Algunos items tienen cambios que aún no se han aprobado."
+    "unsynced_warning": "Algunos items tienen cambios que aún no se han aprobado.",
+    "next": "Siguiente",
+    "total_items": "Total de items para mintear",
+    "confirm_title": "Estas por mintear  {count} {count, plural, un {item} other {items}}, cada item va a ser minteado a la dirección indicada con la cantidad seleccionada.",
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/es.json:1398]: The `{count, plural, one` is part of the logic to pluralize text:
```suggestion
    "confirm_title": "Estas por mintear {count} {count, plural, one {item} other {items}}, cada item va a ser minteado a la dirección indicada con la cantidad seleccionada.",
```

File: src/modules/translation/languages/zh.json [hunk 8]
```diff
@@ -1396,7 +1396,11 @@
     "add_item": "新增项目",
     "limit_reached": "每笔交易只能铸造 {max} 件物品",
     "no_items": "您没有可用于\"{name}\"的项目",
-    "unsynced_warning": "有些项目有尚未批准的更改。"
+    "unsynced_warning": "有些项目有尚未批准的更改。",
+    "next": "下一個",
+    "total_items": "要鑄造的總物品",
+    "confirm_title": "您將要鑄造 {count} {count, plural, 一 {Asset} other {Assets}} 件物品，每件物品都將以提供的數量鑄造到給定的地址。",
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/zh.json:1402]: It seems that in Chinese there's no need to use the pluralized form.
If you translate:
```
You are about to mint 1 item, each item will be minted to the given address with the provided amount.
```
and replace 1 to {count}, it should work fine.

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 9]
```diff
@@ -86,50 +95,98 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
+    const { itemMints, error, confirm } = this.state
 
     const items = this.props.items.concat(this.state.items)
 
     const isEmpty = items.length === 0
     const isFull = items.length === totalCollectionItems
     const isDisabled = this.isDisabled()
 
+    const amountItemsValue = Object.values(itemMints).reduce((accumulator, mints) => {
+      const totalMints = mints.reduce((acc, mint) => {
```
Comment (LautaroPetaccio) [inline src/components/Modals/MintItemsModal/MintItemsModal.tsx:114]:
```suggestion
      const totalMintsPerItem = mints.reduce((acc, mint) => {
```

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 10]
```diff
@@ -86,50 +95,98 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
-    const { itemMints, error } = this.state
+    const { itemMints, error, confirm } = this.state
 
     const items = this.props.items.concat(this.state.items)
 
     const isEmpty = items.length === 0
     const isFull = items.length === totalCollectionItems
     const isDisabled = this.isDisabled()
 
+    const amountItemsValue = Object.values(itemMints).reduce((accumulator, mints) => {
```
Comment (LautaroPetaccio) [inline src/components/Modals/MintItemsModal/MintItemsModal.tsx:113]:
```suggestion
    const totalMints = Object.values(itemMints).reduce((accumulator, mints) => {
```

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 11]
```diff
@@ -139,14 +194,42 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
-            </ModalActions>
-            {error ? (
-              <Row className="error" align="right">
-                <p className="danger-text">{error}</p>
               </Row>
-            ) : null}
-          </Form>
+            </div>
+          ) : (
+            <Form>
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/components/Modals/MintItemsModal/MintItemsModal.tsx:202]: This comment was not answered, would you mind answering it?

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/components/Modals/MintItemsModal/MintItemsModal.css [hunk 12]
```diff
@@ -47,3 +46,37 @@
...
+  align-items: center;
+  gap: 10px;
+  display: flex;
+}
+
+.MintItemsModal .itemImage {
+  min-width: 36px;
+  min-height: 36px;
+  max-width: 36px;
+  max-width: 36px;
```
[the remark this answers is not in the document]
Comment (flobarreto) [reply src/components/Modals/MintItemsModal/MintItemsModal.css:65]: this does not work, is set as this in other parts inside the project

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (flobarreto) [reply src/components/Modals/MintItemsModal/MintItemsModal.tsx:202]: styles looks okay without the form component :)

File: src/components/Modals/MintItemsModal/MintItemsModal.tsx [hunk 13]
```diff
@@ -68,12 +75,14 @@ export default class MintItemsModal extends React.PureComponent<Props, State> {
...
   handleAddItems = (item: Item) => {
-    const { items, itemMints } = this.state
-    this.setState({
-      items: [...items, item],
-      itemMints: {
-        ...itemMints,
-        [item.id]: this.buildMints(item)
+    this.setState(prevState => {
+      return {
+        ...prevState,
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/components/Modals/MintItemsModal/MintItemsModal.tsx:75]: Would you mind removing `...prevState` here?

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**